<launch>


  <arg name="odom_topic" default="/ground_truth/state" />
  <arg name="laser_topic" default="/scan" />
  <arg name="global_frame_id" default="map"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="odom_frame"  default="world"/>


  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
    <param name="base_global_planner" value="navfn/NavfnROS"/>
    <rosparam file="$(find coverage_drone)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find  coverage_drone)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />

    <rosparam file="$(find coverage_drone)/param/costmap_local_params.yaml" command="load" />
    <rosparam file="$(find  coverage_drone)/param/costmap_global_params.yaml" command="load" />

    <rosparam file="$(find  coverage_drone)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find  coverage_drone)/param/planner_dwa_local_params.yaml" command="load" />
    <rosparam file="$(find  coverage_drone)/param/planner_navfn_global_params.yaml" command="load" />


    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame)"/>

    <remap from="cmd_vel" to="/cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>


</launch>